### ISSUE TITLE
Issue with template implemention (tcc) files

### ISSUE THREAD

**Issue by konradsa** [clangd/clangd#262]
Can clangd handle tcc files (commonly used extension is tcc)? When opening a tcc file in vscode, clangd pretty much fails to parse the file correctly, since it doesn't have any includes, the includes are defined in the corresponding header file that includes the tcc file.

Is there any way to get clangd to parse such files correctly?

**Comment by kadircet** [comment]
It is quite similar to https://github.com/clangd/clangd/issues/257

**Comment by konradsa** [comment]
On a second look, it seems to be more of an issue with the namespace than the includes (even though includes could still be an issue).

So let's say the header looks like this:

foo.H:
namespace bar {
...
#include "foo.tcc"
} 

Now foo.tcc implicitly ends up in the bar namespace, but when clangd parses it, it doesn't see that and of course the parsing fails.

Looks like this may be closer to https://github.com/clangd/clangd/issues/45 and quite a tricky issue to solve. These files could never be parsed in isolation, but only together with their header file.

**Comment by konradsa** [comment]
Yes, confirmed, the problem is both the includes as well as the namespace. I think a proper solution would involve finding the corresponding header and parse the tcc file as if it was expanded into the header where the corresponding #include directive is.
